Add prometheus to logging stack

m3dwards opened the pull request:
To test:

```shell
just start(d)
warcli network start
just connectlogging
```

Browse to localhost:3000 (username: admin, password: password)

See that we have Loki and Prometheus data sources already configured and that we have data for Loki and Prometheus. Prometheus bitcoin metrics can take a couple of minutes to appear.

m3dwards committed:
added prometheus to logging stack

m3dwards committed:
update admin password on grafana to be password

m3dwards committed:
delete service monitors when downing network

m3dwards committed:
fix ruff

m3dwards committed:
add prometheus helm chart

m3dwards committed:
select correct continaer for greping logs

willcl-ark commented on src/backends/kubernetes/kubernetes_backend.py:
```diff
@@ -415,6 +420,37 @@ def create_bitcoind_container(self, tank) -> client.V1Container:
+
+    def apply_prometheus_scrape_configs(self, tanks):
+        for tank in tanks:
+            if not tank.exporter:
```
Really like how this is only for enabled nodes 👍🏼

willcl-ark commented on src/backends/kubernetes/kubernetes_backend.py:
```diff
@@ -415,6 +420,37 @@ def create_bitcoind_container(self, tank) -> client.V1Container:
+                continue
+
+            bitcoind_service_name = self.get_service_name(tank.index)
+            target = f"{bitcoind_service_name}.{self.namespace}:9332"
```
in a4d4ee1319cd135f519210a2be75924cd05dc5cd "Add prometheus to logging stack":
can this `9332` magic number be extracted into a variable somewhere? It's also used on line 554.

m3dwards replied:
Extracted

willcl-ark commented on src/backends/kubernetes/kubernetes_backend.py:
```diff
@@ -415,6 +420,37 @@ def create_bitcoind_container(self, tank) -> client.V1Container:
+                "apiVersion": "monitoring.coreos.com/v1alpha1",
+                "kind": "ScrapeConfig",
+                "metadata": {"name": "test", "namespace": "prometheus"},
+                "spec": {"scrapeInterval": "5s", "staticConfigs": [{"targets": [target]}]},
```
in a4d4ee1319cd135f519210a2be75924cd05dc5cd "Add prometheus to logging stack":
perhaps (in the future) this `5s` constant should be configurable in the graph? Unclear to me currently.

m3dwards replied:
Scrapeconfig is no longer used, bad (no) squashing on my part.

willcl-ark commented on src/backends/kubernetes/kubernetes_backend.py:
```diff
@@ -487,8 +523,11 @@ def create_pod_object(
             ),
         )
 
+    def get_service_name(self, tank_index: int) -> str:
```
Glad to see this here, I have it as an unrelated commit in a different branch. Hope to use this instead

willcl-ark commented on src/backends/kubernetes/kubernetes_backend.py:
```diff
@@ -557,6 +597,8 @@ def deploy_pods(self, warnet):
             bitcoind_pod = self.create_pod_object(
                 tank, bitcoind_container, self.get_pod_name(tank.index, ServiceType.BITCOIN)
             )
+            prometheus_container = self.create_prometheus_container(tank)
```
in a4d4ee1319cd135f519210a2be75924cd05dc5cd "Add prometheus to logging stack":

We are only scraping selected containers per the filter on `tank.exporter`:

```python
    def apply_prometheus_scrape_configs(self, tanks):
        for tank in tanks:
            if not tank.exporter
```

Would it not make more sense to not add a prometheus container to them under the same condition?

m3dwards replied:
Added this

willcl-ark commented on src/backends/kubernetes/kubernetes_backend.py:
```diff
@@ -26,7 +26,7 @@
 BITCOIN_CONTAINER_NAME = "bitcoin"
 LN_CONTAINER_NAME = "ln"
-LOGGING_NAMESPACE = "warnet-logging"
+MAIN_NAMESPACE = "warnet"
```
in 95610e44f38e2210d93fdffac73cc51a0ae0ed97 Add prometheus to logging stack:

this is added in the previous commit then removed here?

m3dwards replied:
squashed

willcl-ark commented on src/backends/kubernetes/kubernetes_backend.py:
```diff
@@ -629,8 +637,9 @@ def deploy_pods(self, warnet):
-            self.apply_prometheus_scrape_configs(warnet.tanks)
+
+        # add metrics scraping for tanks configured to export metrics
+        self.apply_prometheus_service_monitor(warnet.tanks)
```
in 95610e44f38e2210d93fdffac73cc51a0ae0ed97 Add prometheus to logging stack:

There seems to be many things modified here which were only added in the previous commit. Perhaps just squash them together? Alternatively just leave; it's not worth splitting and rebasing in any case.

m3dwards replied:
squashed them together

willcl-ark reviewed (commented):
Changes look pretty much great! Nice work.

I did notice quite a few renames/re-touches between the first and second commits, not sure if its just worth squashing those?

One main Q regarding only starting node-exporter for those nodes which prometheus is configured for, but otherwise a tACK from me

m3dwards committed:
only apply prom exporter if metrics are enabled

m3dwards commented:
@willcl-ark I've addressed all your comments.

I still want to make the logging stack optional but that can come in on it's own PR if this gets merged first.

willcl-ark commented:
ACK 1850e31a9ee435ae4ef36256ab7ba758fe41b3e7

Working nicely now, only on configured pods.
